### PR TITLE
bep173: enable text proposal on BNB Smart Chain

### PR DESCRIPTION
**Pull request by randyahx**
## Motivation

In some scenarios, the community may need a proposal that does not directly cause any changes, like agreeing to a certain strategy, plan, commitment or other statement.

## Specification

Right now, there are many system parameters that control the behavior of the BSC, e.g. slash amount, cross-chain transfer fees. All these parameters can be determined by the BSC Validator Set together through a proposal-vote process based on staking. The process executes on the BC and the new parameter values will be picked up by either the management module on the BC or corresponding system contracts via the cross-chain communication protocol. The proposals can be classified into two groups: 

1. Param Change Proposal if the parameter takes effect on the Beacon Chain; 

2. Cross Param Change Proposal if the parameter takes effect on the BNB Smart Chain.

This proposal introduces a new kind of proposal to agree on a certain strategy, plan, commitment, future upgrade, or other statements. **Text proposals** are exclusively a signaling mechanism and focal point for future coordination - they do not directly cause any changes on-chain.

## Commands:

Submit Text Proposal

./tbnbcli gov submit-proposal --type text …[flags]

Query Proposal

./tbnbcli gov query-proposal --proposal-id {id} ...[flags]

Voting 

./tbnbcli gov vote --option yes --proposal-id {id} ...[flags]

### CODE REVIEW

File: asset/testnet/app.toml [hunk 1]
```diff
@@ -60,6 +60,8 @@ BEP128Height = 23551600
 BEP151Height = 28250000
 #Block height of BEP153 upgrade
 BEP153Height = 30516226
+#Block height of BEP173 upgrade
+BEP173Height = 30516226
```
Comment (guagualvcha) [inline asset/testnet/app.toml:64]: need to be decide

File: cmd/gen_devnet/main.go [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+	err := os.RemoveAll(devNetHomeDir)
+	if err != nil {
+		panic(err)
+	}
+	// init nodes
+	cdc := app.Codec
+	ctx := app.ServerContext
+	appInit := app.BinanceAppInit()
+	ctxConfig := ctx.Config
+	sdkConfig := sdk.GetConfig()
```
Comment (guagualvcha) [inline cmd/gen_devnet/main.go:38]: what is this for?

File: common/upgrade/upgrade.go [hunk 3]
```diff
@@ -40,6 +40,7 @@ const (
 	BEP128 = sdk.BEP128 // https://github.com/bnb-chain/BEPs/pull/128 Staking reward distribution upgrade
 	BEP151 = "BEP151"   // https://github.com/bnb-chain/BEPs/pull/151 Decommission Decentralized Exchange
 	BEP153 = sdk.BEP153 // https://github.com/bnb-chain/BEPs/pull/153 Native Staking
+	BEP173 = sdk.BEP173 // https://github.com/bnb-chain/BEPs/pull/173 Text Proposal
```
Comment (guagualvcha) [inline common/upgrade/upgrade.go:43]: Please change the cosmos dep so that the compile can pass.

**Comment by Pythonberg1997** [comment]
Please delete the genDevnet script. And the pr's description needs to be changed. What's `--home ~/go/src/github.com/bnb-chain/node/build/devnet/node0/testnodecli`?

**Review by realuncle — APPROVED** [review]

**Review by guagualvcha — APPROVED** [review]

**Review by Pythonberg1997 — APPROVED** [review]

**Pull request merged**